conda-store-server/tests/test_usage.py

```diff
@@ -0,0 +1,32 @@
+from conda_store_server.utils import disk_usage, du
+
+# TODO: Add tests for the other functions in utils.py
+
+def test_disk_usage(tmp_path):
+    test_dir = tmp_path / "test_dir"
+    test_dir.mkdir()
+
+    # This varies across OSes
+    dir_size = du(test_dir)
+    assert abs(dir_size - int(disk_usage(test_dir))) <= 1000
+
+    test_file = test_dir / "test_file"
+    test_file.write_text("a"*1000)
+    test_file2 = test_dir / "test_file2"
+    test_file2.write_text("b"*1000)
+    # Test hard links
+    test_file_hardlink = test_dir / "test_file_hardlink"
+    test_file_hardlink.hardlink_to(test_file)
+
+    # The exact values depend on the block size and other details. Just check
+    # that it is in the right range. Note that disk_usage uses the du command
+    # on Mac and Linux but uses the Python du() function on Windows.
+    val = disk_usage(test_file)
+    assert isinstance(val, str)
+    assert 1000 <= int(val) <= 1500
+    assert 1000 <= du(test_file) <= 1500
+
+    val = disk_usage(test_dir)
+    assert isinstance(val, str)
+    assert 2000 + dir_size <= int(val) <= 2700 + dir_size
+    assert 2000 + dir_size <= du(test_dir) <= 2700 + dir_size
```

Comment:
This could use more tests. Bounds are 50% of file size, too.

Reply:
Bounds:

```
ubuntu

2023-10-22T18:43:40.3618723Z tests/test_usage.py XXXXXXXXXXXXXX: 1000 1000
2023-10-22T18:43:40.3619644Z XXXXXXXXXXXXXX: 4096 6096 6096

macos

2023-10-22T18:49:11.6388490Z tests/test_usage.py XXXXXXXXXXXXXX: 1024 1000
2023-10-22T18:49:11.6389010Z XXXXXXXXXXXXXX: 64 2560 2160

windows

2023-10-22T18:56:13.8420208Z tests\test_usage.py XXXXXXXXXXXXXX: 1000 1000
2023-10-22T18:56:13.8420808Z XXXXXXXXXXXXXX: 0 2000 2000


```

Reply:
I'll keep this as is right now, but it might be worth it to revise later. Especially, on macOS.

Reply:
> The exact values depend on the block size and other details

Oh yeah this is a fun detail :smile: guess you can't be smaller than the blocksize and also can only take up an integer number of block sizes.